tests/simulate_data/test_get_simulated_data.py

```diff
@@ -0,0 +1,354 @@
+        )
+
+
+# dim_m<0 : l123 ; ValueError: negative dimensions are not allowed
+@pytest.mark.xfail
+def test_dim_m_negative_should_fail():
+    with pytest.raises(Exception):
+        simulate_data(
+            n=10,
+            rg=default_rng(42),
+            mis_spec_m=False,
+            mis_spec_y=False,
+            dim_x=1,
+            dim_m=-1,
+            seed=1,
+            type_m="binary",
+            sigma_y=0.5,
+            sigma_m=0.5,
+            beta_t_factor=1,
+            beta_m_factor=1,
+        )
+
+
+# dim_m>1 ; n>1 ; "binary" ; l39
+@pytest.mark.xfail
+def test_m_multidimensional_binary_works():
+    try:
+        simulate_data(
+            n=7,
+            rg=default_rng(42),
+            mis_spec_m=False,
+            mis_spec_y=False,
+            dim_x=1,
+            dim_m=3,
+            seed=1,
+            type_m="binary",
+            sigma_y=0.5,
+            sigma_m=0.5,
+            beta_t_factor=1,
+            beta_m_factor=1,
+        )
+    except ValueError as err:
+        pprint(err)
+        assert False
+    else:
+        pass
+
+
+# dim_m>1 ; n=1 ; l58
+@pytest.mark.xfail
+def test_m_multidimensional_binary_works1():
+    try:
+        simulate_data(
+            n=1,
+            rg=default_rng(42),
+            mis_spec_m=False,
+            mis_spec_y=False,
+            dim_x=1,
+            dim_m=2,
+            seed=1,
+            type_m="binary",
+            sigma_y=0.5,
+            sigma_m=0.5,
+            beta_t_factor=1,
+            beta_m_factor=1,
+        )
+    except ValueError as err:
+        pprint(err)
+        assert False
+    else:
+        pass
+
+
+# sigma_m grand ; "continuous" ; P(T=1|X,M) = NaN
+@pytest.mark.xfail
+def test_huge_sigma_m_makes_nan():
+    with pytest.raises(Warning):
+        data_temp = simulate_data(
+            n=1,
+            rg=default_rng(42),
+            mis_spec_m=True,
+            mis_spec_y=False,
+            dim_x=1,
+            dim_m=1,
+            seed=1,
+            type_m="fds",
+            sigma_y=0.5,
+            sigma_m=5351,
+            beta_t_factor=1,
+            beta_m_factor=1,
+        )
+    assert data_temp[10] != np.nan
+
+
+# sigma_m=0 ; "continuous" ; P(T=1|X,M) = NaN
+@pytest.mark.xfail
+def test_null_sigma_m_makes_nan():
+    with pytest.raises(Exception):
+        data_temp = simulate_data(
+            n=1,
+            rg=default_rng(42),
+            mis_spec_m=False,
+            mis_spec_y=False,
+            dim_x=1,
+            dim_m=1,
+            seed=1,
+            type_m="continuous",
+            sigma_y=0.5,
+            sigma_m=0,
+            beta_t_factor=1,
+            beta_m_factor=1,
+        )
+    assert data_temp[10] != np.nan
+
+
+# TESTS IMPLEMENTES
+# total = indirect + direct (direct 0 et indirect 1)
+# mis_spec_y=False : 5=6=gamma_t ;  7=8
+# mis_spec_y=True : 5!=6!=gamma_t ;  7!=8
+# vérifier la binarité, la continuité, la dimension de x,t,m,y
```

Comment:
Please avoid writing anything in French. We want this to be readable by anybody.

Reply:
I removed the french comments